astropy/io/fits/tests/test_header.py

```diff
@@ -715,54 +553,54 @@ def test_hierarch_create_and_update(self):
 
```

Comment:
About the number of warnings that changed in these tests: it seems that the old update behavior was not triggering "HIERARCH" warnings, though the new behavior does.